src/components/forms/NumberField/NumberField.js

```diff
@@ -13,10 +13,19 @@ import {NumericFormat} from 'react-number-format';
 
 import {getBEMClassName} from 'utils';
 
+const getSeparators = locale => {
+  const numberFormat = new Intl.NumberFormat(locale);
+  const parts = numberFormat.formatToParts(1000.1);
+  const decimalSeparator = parts.find(part => part.type === 'decimal').value;
+  const thousandSeparator = parts.find(part => part.type === 'group').value;
+  return {decimalSeparator, thousandSeparator};
+};
+
 const NumberField = ({
   name,
   min,
   step,
+  useNumberType = false, // number type can be convenient for small numbers with the +- spinner buttons
```

Comment:
Basically the idea is: the place where the `NumberField` would be used knows best whether `type="number"` or free-form input is best, so let's provide a prop instead of guessing that from other props.

E.g. the product unit would definitely used `<NumberField name="amount" useNumberType min={1} step={1} />` because it's about a discrete number of products you're ordering.